core/templates/pages/exploration-editor-page/statistics-tab/statistics-tab.directive.ts

```diff
@@ -131,169 +147,136 @@ angular.module('oppia').directive('statisticsTab', [
               ];
             });
           };
-          ctrl.onClickStateInStatsGraph = function(stateName) {
-            if (!stateStatsModalIsOpen) {
-              stateStatsModalIsOpen = true;
+
+          ctrl.onClickStateInStatsGraph = stateName => {
+            if (!ctrl.stateStatsModalIsOpen) {
+              ctrl.stateStatsModalIsOpen = true;
               ctrl.showStateStatsModal(
                 stateName, ctrl.highlightStates[stateName]);
             }
           };
 
-          ctrl.showStateStatsModal = function(stateName, improvementType) {
+          ctrl.showStateStatsModal = (stateName, improvementType) => {
+            const state = ExplorationStatesService.getState(stateName);
             AlertsService.clearWarnings();
 
-            StateInteractionStatsService.computeStats(
-              ExplorationStatesService.getState(stateName)
-            ).then(stats => {
+            StateInteractionStatsService.computeStats(state).then(stats => {
               $uibModal.open({
                 templateUrl: UrlInterpolationService.getDirectiveTemplateUrl(
                   '/pages/exploration-editor-page/statistics-tab/templates/' +
                   'state-stats-modal.template.html'),
                 backdrop: true,
                 resolve: {
-                  stateName: function() {
-                    return stateName;
-                  },
-                  stateStats: function() {
-                    return ctrl.stateStats[stateName];
-                  },
-                  improvementType: function() {
-                    return improvementType;
-                  },
-                  visualizationsInfo: function() {
-                    return stats.visualizations_info;
-                  }
+                  customizationArgs: (
```

Comment:
This actually breaks a bunch of conventions, because throughout the rest of the codebase, customizationArgs is typically a dict. Can we preserve that?

Reply:
I've changed the name to `interactionArgs`.